modin/experimental/core/storage_formats/pandas/native_query_compiler.py

```diff
@@ -0,0 +1,1283 @@
+    if inplace:
+        df.sort_index(inplace=inplace, **kwargs)
+    else:
+        df = df.sort_index(inplace=inplace, **kwargs)
+    return df
+
+
+def _combine(df, other, func, **kwargs):  # noqa: GL08
+    if isinstance(df, pandas.Series):
+        return func(df, other)
+    return df.combine(other, func)
+
+
+def _getitem_array(df, key):  # noqa: GL08
+    if isinstance(key, pandas.DataFrame):
+        key = key.squeeze(axis=1)
+    return df[key]
+
+
+def _getitem_row_array(df, key):  # noqa: GL08
+    if isinstance(key, pandas.DataFrame):
+        key = key.squeeze(axis=1)
+    return df.iloc[key]
+
+
+def _write_items(
+    df,
+    row_numeric_index,
+    col_numeric_index,
+    item,
+    need_columns_reindex=True,
+):  # noqa: GL08
+    from modin.pandas.utils import broadcast_item, is_scalar
+
+    if not isinstance(row_numeric_index, slice):
+        row_numeric_index = list(row_numeric_index)
+    if not isinstance(col_numeric_index, slice):
+        col_numeric_index = list(col_numeric_index)
+    if not is_scalar(item):
+        broadcasted_items, _ = broadcast_item(
+            df,
+            row_numeric_index,
+            col_numeric_index,
+            item,
+            need_columns_reindex=need_columns_reindex,
+        )
+    else:
+        broadcasted_items = item
+
+    if isinstance(df.iloc[row_numeric_index, col_numeric_index], pandas.Series):
+        broadcasted_items = broadcasted_items.squeeze()
+    df.iloc[row_numeric_index, col_numeric_index] = broadcasted_items
+    return df
+
+
+def _setitem(df, axis, key, value):  # noqa: GL08
+    if is_scalar(key) and isinstance(value, pandas.DataFrame):
+        value = value.squeeze()
+    if not axis:
+        df[key] = value
+    else:
+        df.loc[key] = value
+    return df
+
+
+def _delitem(df, key):  # noqa: GL08
+    return df.drop(columns=[key])
+
+
+def _get_dummies(df, columns, **kwargs):  # noqa: GL08
+    return pandas.get_dummies(df, columns=columns, **kwargs)
+
+
+def _register_default_pandas(
+    func,
+    is_series=False,
+    squeeze_series=False,
+    squeeze_args=False,
+    squeeze_kwargs=False,
+    return_modin=True,
+    in_place=False,
+    df_copy=False,
+    filter_kwargs=[],
+):
+    """
+    Build function that apply specified method of the passed frame.
+
+    Parameters
+    ----------
+    func : callable
+        Function to apply.
+    is_series : bool, default: False
+        If True, the passed frame will always be squeezed to a series.
+    squeeze_series : bool, default: False
+        If True, the passed frame will always be squeezed to a series if there is a single column named "__reduced__".
+    squeeze_args : bool, default: False
+        If True, all passed arguments will be squeezed.
+    squeeze_kwargs : bool, default: False
+        If True, all passed key word arguments will be squeezed.
+    return_modin : bool, default: True
+        If True, the result will always try to convert to DataFrame or Series.
```

Comment:
```suggestion
        If False, the result will always try to convert to DataFrame or Series.
```
Should this be False instead?

Reply:
No it is supposed to be True, in case if it is false and the result is not Dataframe or Series we return result as is without converting to Dataframe,

 The purpose of this is to be used to set this argument to false for functions like mean which would return a scalar, and conversion to dataframe is not required.

Reply:
I get it now, thanks! However, it seemed to me a little confusing. Maybe we should rename this parameter to `return_raw` and set it to True for functions like mean?

Reply:
Done.